### ISSUE TITLE
An action without parameters results in NoSuchMethodException starting with Spring Framework 6.2.0-M1

### ISSUE THREAD

**Issue by transentia** [spring-projects/spring-webflow#1802]
Long-established app. uses Webflow 3.0.0.
Thought I'd try the upgrade to Framework 6.2.0-SNAPSHOT...see what breaks.

This code broke:

```
public class ExistingReportActions extends FormAction {
    ...

    public Event evaluateExistingReport() {
        ...
    }
```

Rectified by supplying the 'missing' RequestContext parameter explicitly:

```
public class ExistingReportActions extends FormAction {
    ...

    public Event evaluateExistingReport(RequestContext requestContext) {
        ...
    }
```

The flow xml remains as:

```
    <!-- *** existingReport MUST BE THE FIRST STATE in the flow *** -->
    <action-state id="existingReport">
        <evaluate expression="existingReportActions.evaluateExistingReport"/>
        ....
    </action-state>
```

Eliding the RequestContext parameter used to be allowable. It appears that it no longer is.

A fairly trivial difference, but a breaking one nonetheless.

**Comment by rstoyanchev** [comment]
Thanks for the report. Could you extract a small reproducer?

**Comment by rstoyanchev** [comment]
In 6.0, the method is called successfully via SpEL, but in 6.2 the same expression does not return a value, and Web Flow tries the overloaded method signature with a `RequestControl` parameter, and fails if that doesn't exist. 

It does work with parenthesis, i.e. "existingReportActions.evaluateExistingReport()".

**Comment by rstoyanchev** [comment]
This changed in 6.2.0-M1. I cannot reproduce it in 6.1.x. 

I have not yet narrowed it down to a specific change, but suspect it is expected, and we may just have to document it as an upgrade note.

**Comment by sbrannen** [comment]
I'm currently investigating whether we can support the previous use case in Spring Web Flow, but in the interim I can provide my initial findings here.

Basically, there are two different SpEL features that come into play here.

- property reference
- method reference

`existingReportActions.evaluateExistingReport` is a property reference that is supported by a registered `PropertyAccessor`.

`existingReportActions.evaluateExistingReport()` and `existingReportActions.evaluateExistingReport(requestContext)` are both method references.

Thus, adding the `()` to `existingReportActions.evaluateExistingReport` _converts_ it to a method reference which is handled by `org.springframework.expression.spel.ast.MethodReference` instead of `org.springframework.expression.spel.ast.PropertyOrFieldReference`.

----

Without the `()`, a registered `PropertyAccessor` is used to _invoke_ the property accessor method.

If the `org.springframework.expression.spel.support.DataBindingPropertyAccessor` is used (which appears to be the case prior to Spring Framework 6.2), the invocation works. 

However, if the `org.springframework.webflow.expression.spel.ActionPropertyAccessor` is used (which appears to be the case starting with Spring Framework 6.2 M1), the invocation fails since the method does not declare a `RequestContext` argument which is required by `org.springframework.webflow.action.MultiAction.doExecute(RequestContext)`.

I believe the change in behavior (switch from `DataBindingPropertyAccessor` to `ActionPropertyAccessor`) is due to the following improvements for `PropertyAccessor` sorting.

- https://github.com/spring-projects/spring-framework/commit/4b0a04857004e3d0e1040a1e70d6ac78af086a6f
- https://github.com/spring-projects/spring-framework/commit/d91277095a3eee690f3e6766257cec6e1c2510f8

In other words, I believe the behavior with Spring Framework 6.2 is the expected behavior in terms of SpEL contracts, and I believe those fixes/improvements in SpEL highlight something that maybe never should have worked in Spring Web Flow with the current implementations of `ActionPropertyAccessor` and `MultiAction`.

I will attempt to confirm that last statement and report back here.

**Comment by sbrannen** [comment]
> I will attempt to confirm that last statement and report back here.

I confirmed that.

#### Spring Framework 6.1

evaluateFoo:

- ReflectivePropertyAccessor
- FlowVariablePropertyAccessor
- BeanFactoryPropertyAccessor
- ActionPropertyAccessor

#### Spring Framework 6.2

evaluateFoo:

- ActionPropertyAccessor
- ReflectivePropertyAccessor
- FlowVariablePropertyAccessor
- BeanFactoryPropertyAccessor

----

Thus, `ActionPropertyAccessor` is now correctly queried **before** `ReflectivePropertyAccessor` with Spring Framework 6.2.

The following change to `ActionPropertyAccessor` allows the reported use case to work.

```java
@Override
public boolean canRead(EvaluationContext context, Object target, String name) {
	if (!(target instanceof Action)) {
		return false;
	}
	Method method = ReflectionUtils.findMethod(target.getClass(), name, RequestContext.class);
	return (method != null);
}
```

@rstoyanchev, I'm happy to put together a PR for this. Otherwise, we can discuss options offline.
